pages/common/xargs.md

```diff
@@ -1,13 +1,17 @@
 # xargs
 
 > Execute a command with piped arguments coming from another command, a file, etc.
-> The input is treated as a single block of text and split into separate arguments on spaces, tabs, newlines and end-of-file.
+> The input is treated as a single block of text and split into separate pieces on spaces, tabs, newlines and end-of-file.
 
-- Main usage pattern:
+- Main usage pattern — run a command once for each piece of the input data:
```

Comment:
```suggestion
- Run a command using the input data as arguments:
```

Reply:
On my machine:

|                                 | `xargs echo`      | `xargs -I echo`   |
|---------------------------------|-------------------|-------------------|
| `echo a b c`                    | `a b c`           | `<empty line>`    |
| `echo -e 'a\tb\tc'`             | `a b c`           | `<empty line>`    |
| `echo -e 'a\nb\nc'`             | `a b c`           | `<3 empty lines>` |


I'm not quite sure what's the point of `xargs -I echo` really.

Nonetheless, our description of `xargs` splitting incoming arguments on any whitespace seems to be right, and this is confirmed both on your and my machine. 

It does not run a command for each argument, it just splits the input and uses each part as an argument, running one single command.

---

    $ echo 'a b c' | strace -f xargs echo | grep exec | grep unfinish
    [pid 26656] execve("/bin/echo", ["echo", "a", "b", "c"], [/* 40 vars */] <unfinished ...>

    $ echo 'a\tb\tc' | strace -f xargs echo | grep exec | grep unfinish
    [pid 26656] execve("/bin/echo", ["echo", "a", "b", "c"], [/* 40 vars */] <unfinished ...>

    $ echo 'a\nb\nc' | strace -f xargs echo | grep execve
    [pid 26656] execve("/bin/echo", ["echo", "a", "b", "c"], [/* 40 vars */] <unfinished ...>

    $ echo 'a b c' | 
    [pid 26867] execve("/bin/echo", ["echo"], [/* 40 vars */] <unfinished ...>

Reply:
Yeah, I tried on my Linux machine at home and it looks like the placeholder is required for `xargs -I`, like this: `echo -n 'a\nb\nc' | xargs -I _ echo _`. With that syntax, I can reproduce the same results I got on macOS. I'm out of time today to dig deeper into this, will follow up later.

Reply:
Ok, `xargs -I _` seems to pass all the input as a single argument replacing the `_` placeholder.

|                     | `xargs -I _ echo _`   |
|-------------------- |-----------------------|
| `echo a b c`        | <pre>a b c</pre>  |
| `echo -e 'a\tb\tc'` | <pre>a	b	c</pre> |
| `echo -e 'a\nb\nc'` | <pre>a<br>b<br>c<br></pre> |

So you want to add an example for this, I suppose?

Reply:
Hmm. I may have let too much time pass since I first opened the PR and maybe lost the proper context, but indeed re-reading this conversation I'm not really sure why I threw `xargs -I` into the mix here 😅.

Getting back to the point, your edit suggestion is of course correct for the main usage (without `-I`) of the command. I'll make the change.

> So you want to add an example for this, I suppose?

We already have an example for `xargs -I` on this page (it's the second-to-last one), and the description there is indeed consistent with what we found here: it splits the input on newlines and runes one command for each line. So there's nothing to do there :)